lib/msf/core/exploit/remote/http_client.rb

```diff
@@ -286,6 +264,38 @@ def connect_ws(opts={}, timeout = 20)
     socket.extend(Rex::Proto::Http::WebSocket::Interface)
   end
 
+
+  # Returns a proc to log HTTP requests and responses
+  # if datastore['HttpTrace'] is set to true.
+  def set_http_trace_proc(http_trace)
+    if datastore['HttpTrace']
+      proc_httptrace = proc { |request, response| 
+        request_color, response_color = 
+          (datastore['HttpTraceColors'] || 'red/blu').split('/').map { |color| "%bld%#{color}" }
+        
+        request = request.to_s(headers_only: datastore['HttpTraceHeadersOnly'])
+        print_line('#' * 20)
+        print_line('# Request:')
+        print_line('#' * 20)
+        print_line("%clr#{request_color}#{request}%clr")
+
+        print_line('#' * 20)
+        print_line('# Response:')
+        print_line('#' * 20)
+        
+        if response
+          response = response.to_terminal_output(headers_only: datastore['HttpTraceHeadersOnly'])
+          print_line("%clr#{response_color}#{response}%clr")
+        else
+          print_line('No response received')
+        end
+      }
+    end
+
+   proc_httptrace 
```

Comment:
This will be undefined if `datastore['HttpTrace']` is a false value such as `nil`.